server/service/queries.go

```diff
@@ -317,6 +317,13 @@ func (svc *Service) NewQuery(ctx context.Context, p fleet.QueryPayload) (*fleet.
 	if p.DiscardData != nil {
 		query.DiscardData = *p.DiscardData
 	}
+	if len(p.LabelsIncludeAny) > 0 {
+		labelIdents := make([]fleet.LabelIdent, 0, len(p.LabelsIncludeAny))
+		for _, label := range p.LabelsIncludeAny {
+			labelIdents = append(labelIdents, fleet.LabelIdent{LabelName: label})
+		}
+		query.LabelsIncludeAny = labelIdents
+	}
```

Comment:
We need to add this to [ModifyQuery](https://github.com/fleetdm/fleet/blob/bdaf43df6e027124db77c43f0f269349b4b67514/server/service/queries.go#L381) as well

Reply:
(modified to handle the case of no labels, so they can clear all labels from a query)

Reply:
Oh shizzle you right!